presto-cli/src/main/java/io/prestosql/cli/TableNameCompleter.java

```diff
@@ -58,10 +58,18 @@ public TableNameCompleter(QueryRunner queryRunner)
 
     private List<String> listTables(String schemaName)
     {
-        return queryMetadata(format("SELECT table_name FROM information_schema.tables WHERE table_schema = '%s'", schemaName));
+        String[] array = schemaName.split("\\.");
+        if (array.length == 1) {
+            return queryMetadata(format("SELECT table_schema || '.' || table_name FROM %s.information_schema.tables", array[0]));
+        }
+        else if (array.length == 2) {
+            return queryMetadata(format("SELECT table_name FROM information_schema.tables WHERE table_schema = '%s'", array[1]));
```

Comment:
Would it be possible to use prepared statement for passing the schema?

Otherwise you'd need to escape apostrophes in schema name.

Reply:
Fixed to escape.